### PR TITLE
Add 'Subscription.set_iam_policy' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1644 as a base.~~

Toward #1073.

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merging #1644 to `master`.  PTAL

File: docs/pubsub-usage.rst [hunk 1]
```diff
@@ -329,3 +329,15 @@ Fetch the IAM policy for a subscription
...
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> subscription = topic.subscription('subscription_name')
+   >>> policy = subscription.get_iam_policy()  # API request
+   >>> policy.writers.add(policy.group('editors-list@example.com'))
+   >>> subscriptoin.set_iam_policy(policy)  # API request
```
Comment (dhermes) [inline docs/pubsub-usage.rst]: s/`subscriptoin`/`subscription`

File: gcloud/pubsub/subscription.py [hunk 2]
```diff
@@ -283,3 +283,28 @@ def get_iam_policy(self, client=None):
...
+
+    def set_iam_policy(self, policy, client=None):
+        """Update the IAM policy for the subscription.
+
+        See:
+        https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/setIamPolicy
+
+        :type policy: :class:`gcloud.pubsub.iam.Policy`
+        :param policy: the new policy, typically fetched via
+                       :meth:`getIamPolicy` and updated in place.
```
Comment (dhermes) [inline gcloud/pubsub/subscription.py]: `getIamPolicy` is not a method on this class

File: gcloud/pubsub/test_subscription.py [hunk 3]
```diff
@@ -555,6 +555,90 @@ def test_get_iam_policy_w_alternate_client(self):
...
+        WRITER2 = 'user:phred@example.com'
+        READER1 = 'serviceAccount:1234-abcdef@service.example.com'
+        READER2 = 'user:phred@example.com'
+        POLICY = {
+            'etag': 'DEADBEEF',
+            'version': 17,
+            'bindings': [
+                {'role': 'roles/owner', 'members': [OWNER1, OWNER2]},
+                {'role': 'roles/writer', 'members': [WRITER1, WRITER2]},
+                {'role': 'roles/reader', 'members': [READER1, READER2]},
```
Comment (dhermes) [inline gcloud/pubsub/test_subscription.py]: Use the constants?

**Comment by dhermes** [comment]
LGTM other than a few small nits

File: docs/pubsub-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/pubsub-usage.rst]: 054e934

File: gcloud/pubsub/subscription.py (hunk 2 above)
Comment (tseaver) [inline gcloud/pubsub/subscription.py]: 45fcfdf

File: gcloud/pubsub/test_subscription.py (hunk 3 above)
Comment (tseaver) [inline gcloud/pubsub/test_subscription.py]: 8cc3057

**Pull request merged**